db/version_set.h

```diff
@@ -893,6 +904,10 @@ class VersionSet {
   // The last seq that is already allocated. The seq might or might not have
   // appreated in memtable.
   std::atomic<uint64_t> last_allocated_sequence_;
+  // The last allocated sequence that is also published to the readers. This is
+  // used only when allocate_seq_only_for_data_ is not set. Otherwise
+  // last_sequence_ also indicates the last published seq.
+  std::atomic<uint64_t> last_published_sequence_;
```

Comment:
I still feel LastPublishedSequence can combined with LastSequence. Can you explain again what's the reason to split them?

Reply:
I first implemented that approach and got some test failures. I did not quite figure the reason for the test failure but the process of figuring was complicated since last_sequence indicating the last sequence present in the memtable is a fundamental assumption in the code base. At this late stage of the project we cannot afford the risk of modifying a principal assumption in the code base and deal with the probable bugs later.

Reply:
make sense. So summary:
last_allocated_sequence: last sequence consumed by writers but not necessary land in memtable yet.
last_sequence: last sequence finish writting to memtable
last_published_sequence: last sequence that should be visible to any reader
and last_allocated_sequence <= last_sequence <= last_published_sequence.

Reply:
Well said,  i should add this to the code as a comment.
This is for two write queue case. In one queue case, all the roles are played with last_sequence.